src/cc/lib/graph/metadata.cc

```diff
@@ -37,176 +41,102 @@ Metadata::Metadata(std::filesystem::path path, std::string config_path)
     }
 #endif
 
-    auto meta = open_meta(std::move(path), "rt");
-    if (fscanf(meta, "v%zu\n", &m_version) <= 0)
-    {
-        RAW_LOG_ERROR(
-            "Failed to read binary data version from meta file. Please use latest deepgnn package to convert data.");
-        exit(errno);
-    }
+    std::ifstream f(path / "meta.json");
```

Comment:
What part is not clear?

Reply:
It looks like there is no such checks before this, I will add a check to verify it is json format.

Reply:
> It looks like there is no file exists check before this

Of course there is a check like this, `open_meta` function handles it.

Reply:
I just added the file exists check, since if the .json file does not exist it is out of date. Also it is in the python not cc since that gets caught first.

Reply:
That is true, we don't need a separate check to see if the file is json or not though. It looks like this python code runs no matter what: https://github.com/microsoft/DeepGNN/blob/2b4d5bb6674faefcd24010470ecff3bfeb1fdb2b/src/python/deepgnn/graph_engine/snark/meta.py#:~:text=%2C%20config_path)-,try%3A,),-self.version